packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/account-page-list.tsx

```diff
@@ -0,0 +1,327 @@
+  currentPageEntityId,
+  accountId,
+}) => {
+  const { data, loading: pagesLoading } = useAccountPages(accountId);
+
+  const [createUntitledPage, { loading: createUntitledPageLoading }] =
+    useCreatePage(accountId);
+  const [createSubPage, { loading: createSubpageLoading }] =
+    useCreateSubPage(accountId);
+  const [reorderPage, { loading: reorderLoading }] = useReorderPage(accountId);
+  const [archivePage, { loading: archivePageLoading }] = useArchivePage();
+
+  const [initialLoading, setInitialLoading] = useState(true);
+
+  const loading =
+    pagesLoading ||
+    createUntitledPageLoading ||
+    createSubpageLoading ||
+    reorderLoading ||
+    archivePageLoading;
+
+  const [expandedPageIds, setExpandedPageIds] = useLocalstorageState<string[]>(
+    "hash-expanded-sidebar-pages",
+    [],
+  );
+
+  const [treeItems, setTreeItems] = useState<TreeItem[]>([]);
+
+  const [activeId, setActiveId] = useState<UniqueIdentifier | null>(null);
+  const [overId, setOverId] = useState<UniqueIdentifier | null>(null);
+  const [offsetLeft, setOffsetLeft] = useState(0);
+
+  // @todo handle loading/error states properly
+  const addPage = useCallback(async () => {
+    if (loading) {
+      return;
+    }
+
+    try {
+      await createUntitledPage(getLastIndex(treeItems));
+    } catch (err) {
+      // eslint-disable-next-line no-console -- TODO: consider using logger
+      console.error("Could not create page: ", err);
+    }
+  }, [createUntitledPage, loading, treeItems]);
+
+  const handleToggle = (nodeId: string) => {
+    setExpandedPageIds((expandedIds) =>
+      expandedIds.includes(nodeId)
+        ? expandedIds.filter((id) => id !== nodeId)
+        : [...expandedIds, nodeId],
+    );
+  };
+
+  const collapsedPageIds = useMemo(
+    () =>
+      treeItems
+        .filter((item) =>
+          isPageCollapsed(item, treeItems, expandedPageIds, activeId),
+        )
+        .map(({ page }) => page.entityId),
+    [treeItems, expandedPageIds, activeId],
+  );
+
+  const pagesFlatIdList = useMemo(
+    () => treeItems.map(({ page }) => page.entityId),
+    [treeItems],
+  );
+
+  useEffect(() => {
+    // We only update the state when everything has been loaded.
+    // If the request fails, pages will be reordered to their original state
+    if (!loading) {
+      setTreeItems(getTreeItemList(data));
+      setInitialLoading(false);
+    }
+  }, [data, loading]);
+
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 5,
+      },
+    }),
+  );
+
+  const dragDepth = useMemo(
+    () => Math.round(offsetLeft / IDENTATION_WIDTH),
+    [offsetLeft],
+  );
+
+  const projected =
+    activeId && overId
+      ? getProjection(treeItems, collapsedPageIds, activeId, overId, dragDepth)
+      : null;
+
+  useEffect(() => {
+    document.body.style.setProperty("cursor", activeId ? "grabbing" : "");
+  }, [activeId]);
```

Comment:
This ensures the cursor is restored when this component unmounts, and is also clearer. 

```suggestion
  useEffect(() => {
    if (activeId) {
      document.body.style.setProperty("cursor", "grabbing");
      
      return () => {
        if (document.body.style === "grabbing") {
          document.body.style.setProperty("cursor", "");
        }
      };
    }
  }, [activeId]);
```

Reply:
Done in f066c2f938e05d47d0e1f99195c9addf2293d78a